examples/jsm/lines/LineMaterial.js

```diff
@@ -453,247 +453,181 @@ class LineMaterial extends ShaderMaterial {
+	set color( value ) {
 
-						this.needsUpdate = true;
+		this.uniforms.diffuse.value = value;
 
-					}
+	}
 
-					if ( value === true ) {
+	get worldUnits() {
 
-						this.defines.USE_DASH = '';
+		return 'WORLD_UNITS' in this.defines;
 
-					} else {
+	}
 
-						delete this.defines.USE_DASH;
+	set worldUnits( value ) {
 
-					}
-
-				}
+		if ( value === true ) {
 
-			},
+			this.defines.WORLD_UNITS = '';
 
-			dashScale: {
+		} else {
 
-				enumerable: true,
+			delete this.defines.WORLD_UNITS;
 
-				get: function () {
+		}
 
-					return this.uniforms.dashScale.value;
+	}
 
-				},
+	get linewidth() {
 
-				set: function ( value ) {
+		return this.uniforms.linewidth.value;
 
-					this.uniforms.dashScale.value = value;
+	}
 
-				}
+	set linewidth( value ) {
 
-			},
+		if ( ! this.uniforms.linewidth ) return;
+		this.uniforms.linewidth.value = value;
 
-			dashSize: {
+	}
 
-				enumerable: true,
+	get dashed() {
 
-				get: function () {
+		return 'USE_DASH' in this.defines;
 
-					return this.uniforms.dashSize.value;
+	}
 
-				},
+	set dashed( value ) {
 
-				set: function ( value ) {
+		if ( ( value === true ) !== this.dashed ) {
 
-					this.uniforms.dashSize.value = value;
+			this.needsUpdate = true;
 
-				}
+		}
 
-			},
+		if ( value === true ) {
 
-			dashOffset: {
+			this.defines.USE_DASH = '';
 
-				enumerable: true,
+		} else {
 
-				get: function () {
+			delete this.defines.USE_DASH;
 
-					return this.uniforms.dashOffset.value;
+		}
 
-				},
+	}
 
-				set: function ( value ) {
+	get dashScale() {
 
-					this.uniforms.dashOffset.value = value;
+		return this.uniforms.dashScale.value;
 
-				}
+	}
 
-			},
+	set dashScale( value ) {
 
-			gapSize: {
+		this.uniforms.dashScale.value = value;
 
-				enumerable: true,
+	}
 
-				get: function () {
+	get dashSize() {
 
-					return this.uniforms.gapSize.value;
+		return this.uniforms.dashSize.value;
 
-				},
+	}
 
-				set: function ( value ) {
+	set dashSize( value ) {
 
-					this.uniforms.gapSize.value = value;
+		this.uniforms.dashSize.value = value;
 
-				}
+	}
 
-			},
+	get dashOffset() {
 
-			opacity: {
+		return this.uniforms.dashOffset.value;
 
-				enumerable: true,
+	}
 
-				get: function () {
+	set dashOffset( value ) {
 
-					return this.uniforms.opacity.value;
+		this.uniforms.dashOffset.value = value;
 
-				},
+	}
 
-				set: function ( value ) {
+	get gapSize() {
 
-					this.uniforms.opacity.value = value;
+		return this.uniforms.gapSize.value;
 
-				}
+	}
 
-			},
+	set gapSize( value ) {
 
-			resolution: {
+		this.uniforms.gapSize.value = value;
 
-				enumerable: true,
+	}
 
-				get: function () {
+	get opacity() {
 
-					return this.uniforms.resolution.value;
+		return this.uniforms.opacity.value;
 
-				},
+	}
 
-				set: function ( value ) {
+	set opacity( value ) {
 
-					this.uniforms.resolution.value.copy( value );
+		if ( ! this.uniforms ) return;
+		this.uniforms.opacity.value = value;
```

Comment:
Can you explain the reason for this line, please?

Reply:
Same as above, but `opacity` is set in `Material`.